src/electron/window-frame-listener.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BrowserWindow } from 'electron';
+import { WindowStateActionCreator } from './flux/action-creator/window-state-action-creator';
+
+export class WindowFrameListener {
+    constructor(private readonly windowStateActionsCreator: WindowStateActionCreator, private readonly browserWindow: BrowserWindow) {}
+
+    public initialize(): void {
+        this.browserWindow.on('minimize', this.onWindowStateChange);
+        this.browserWindow.on('maximize', this.onWindowStateChange);
+        this.browserWindow.on('unmaximize', this.onWindowStateChange);
+    }
```

Comment:
Looking over the [possible events](https://electronjs.org/docs/api/browser-window#instance-events), do you think we might want to listen to some of these?
`restored`
`enter-full-screen`
`leave-full-screen`

Full screen would only matter if that is the event that gets thrown on MacOS when the maximize equivalent is clicked.

Reply:
i havnt tested in mac. I will test in mac & get back if these events are needed. i don't need these events for windows, for our current usage.

Reply:
i will be sending another pr to address mac os.